scripts/configure

```diff
@@ -262,6 +262,9 @@ APP_AGORA_SERVER_IP="10.21.21.88"
 APP_AGORA_FILEBROWSER_IP="10.21.21.89"
 APP_SPARKKIOSK_IP="10.21.21.90"
 APP_SPARKKIOSK_PORT="21214"
+APP_IPFSHOST_IP=`ip -o route get to 1.1.1.1 | sed -n 's/.*src \([0-9.]\+\).*/\1/p'`
```

Comment:
Unfortunately we can't really do this. This file is just a short term hack for us to hardcode some settings that will in the future be generated on the fly by Umbrel and assigned to an app. We can't allow apps execute arbitrary commands here.

Does the app need to know the public IP to run? If it does, could this logic be done from inside the container?

Reply:
Hmmn, enaling CORS for everything could leak data from the IPFS API unless it implements CSRF protection (maybe it does I'm not familiar with IPFS).

Reply:
I assumed the host IP was basically the same as "*" because everything is coming through the docker/host IP anyway (correct?). i.e. if they have access to the lan/host IP they'll have access to the IPFS API no matter how IPFS/CORS is configured.

Reply:
The attack vector would be CSRF from a malicious website so:

- Trick user to visit malicious website
- Malicious website runs JS that does fetch request to http://umbrel.local:5001/api?get-some-sensitive-ipfs-data
- Due to relaxed CORS policy, website can read sensitive repsonse

So I just drop a link to evil.com in Umbrel telegram and anyone that clicks the link while on their local network leaks IPFS data to me.